hw/adsp/dsp/common.c

```diff
@@ -125,24 +125,25 @@ static int sof_module_memcpy(struct adsp_dev *adsp,
 			continue;	/* not handled atm */
 		case SOF_FW_BLK_TYPE_IRAM:
 			fprintf(stdout, "text: 0x%x size 0x%x\n",
-				board->iram_base + block->offset - board->host_iram_offset,
+				board->mem_zones[block->type].base + block->offset -
+				board->mem_zones[block->type].host_offset,
 				block->size);
 
-			mem = adsp_get_mem_space(adsp, board->iram_base + block->offset - board->host_iram_offset);
+			mem = adsp_get_mem_space(adsp, board->mem_zones[block->type].base + block->offset - board->mem_zones[block->type].host_offset);
 			if (!mem)
 				goto next;
-			memcpy(mem->ptr + block->offset - board->host_iram_offset,
+			memcpy(mem->ptr + block->offset - board->mem_zones[block->type].host_offset,
 				(void *)block + sizeof(*block), block->size);
 			break;
 		case SOF_FW_BLK_TYPE_DRAM:
 			fprintf(stdout, "data: 0x%x size 0x%x\n",
-				board->dram_base + block->offset - board->host_dram_offset,
+				board->mem_zones[block->type].base + block->offset - board->mem_zones[block->type].host_offset,
 				block->size);
 
-			mem = adsp_get_mem_space(adsp, board->dram_base + block->offset - board->host_dram_offset);
+			mem = adsp_get_mem_space(adsp, board->mem_zones[block->type].base + block->offset - board->mem_zones[block->type].host_offset);
 			if (!mem)
 				goto next;
-			memcpy(mem->ptr + block->offset - board->host_dram_offset,
+			memcpy(mem->ptr + block->offset - board->mem_zones[block->type].host_offset,
 				(void *)block + sizeof(*block), block->size);
 			break;
 		default:
```

Comment:
@dianacretu we can further simplify this. Since now we are addressing mem_zones based on block->type, you notice that the code under 'case SOF_FW_BLK_TYPE_IRAM' and 'case SOF_FW_BLK_TYPE_DRAM' is identical so instead of:

```
case SOF_FW_BLK_TYPE_IRAM:
  /* ... */
case SOF_FW_BLK_TYPE_DRAM:
/* .... */

you can say:
case SOF_FW_BLK_TYPE_IRAM:
case SOF_FW_BLK_TYPE_DRAM:
/* ... *.